src/Domain/DonationService.php

```diff
@@ -282,6 +283,18 @@ private function runWithPossibleRetry(\Closure $retryable, string $actionName):
                 $seconds = (new Randomizer())->getFloat(0.1, 1.1);
                 \assert(is_float($seconds)); // See https://github.com/vimeo/psalm/issues/10830
                 $this->clock->sleep($seconds);
+
+                if ($retryCount === self::MAX_RETRY_COUNT) {
+                    $this->logger->error(
+                        sprintf(
+                            $actionName . ' error: %s. Giving up after %d retries.',
+                            $exception->getMessage(),
+                            self::MAX_RETRY_COUNT,
+                        )
+                    );
+
+                    throw $exception;
+                }
```

Comment:
this LGTM - equivalent to wrapping the entire runWithPossibleRetry in another try/catch just to log the final error in case all tries fail, making it clear that it failed after the retries.